### ISSUE TITLE
submission deletion fails for large submissions

### ISSUE THREAD

**Issue by amnonkhen** [ebi-ait/dcp-ingest-central#923]
Submission deletion logic is very inefficient. It finds and unlinks all meta data entities related to this submission by using the following steps for each metadata type, e.g. Protocol:
1. find all entities related to this submission using the relevant repository, e.g. `ProtocolRepository` - a query to the db
1. for each, remove all links using the relevant `MetaDataCrudStrategy.removeLinksToDocument`, e.g. `ProcessCrudStrategy`
1. This calls one or more related repositories to find the documents related to the metadata entity. Each is a find query
1. each result of those turns into a single doc update query in mongo
1. after the links cleanup is done, there is a single call to each repository (9 in total) to delete by the submission.

In total we have:
- a find query per link type, e.g. biomaterial-process. This is not too bad.
- a save query for each link. This has a huge overhead for projects with many links.

The initial 4 steps are only required when the project has more than a single submission. For example:
- submission 1 - contains donor
- submission 2 - contains the specimen which is linked to that donor, that happens thru a new process in submission 2 being added to the `donor.inputToProcesses` property and `specimen.derivedByProcesses` property

In such a case, if we only do step 5, the donor will reference a deleted process id and that can cause problems.

In other words, for projects that only have a single submission, we could skip steps 1-4.

**tasks**
- [ ] link to datasets where it happened
- [ ] add tests that will check the 1 submission use case and 2 submission use case.
- [ ] run all unit tests of core
- [ ] test on dev
